### ISSUE TITLE
XAML designer will not render any forms in these projects?

### ISSUE THREAD

**Issue by GuildOfCalamity** [PacktPublishing/-Learn-WinUI-3.0#1]
![XAML_issue](https://user-images.githubusercontent.com/81915293/125095450-2d0bbd80-e0a2-11eb-93ed-c71c180ef47c.png)

I have installed the WinUI extensions in VS2019 and followed all the steps in the beginning chapter of the book but I cannot get the form to paint in the IDE. I have checked all of the settings in the options menus and my other XAML projects render the form just fine. The only issue I can see is this complaint about the "Microsoft.ApplicationModel.Resources.winmd" in the project references (see screenshot). NOTE: The app will run and compile without issue, but will not render in the IDE.

Anyone have any ideas?

**Comment by alvinashcraft** [comment]
I'll take a look at it this weekend. The samples were checked to be functional at the time of publication in March, but there have been many changes from Microsoft in the time since then.

**Comment by GuildOfCalamity** [comment]
Thank you. I did notice that when you install WinUI 3.0 in VS2019 it also forces the Project Reunion install, so maybe there's some order of operations to the installation of the components that I did in the wrong order, but I didn't see any install errors from Visual Studio.

I tried doing a "dotnet restore" on the project but that did not fix the reference.

When I look into the Microsoft.ApplicationModel.Resources namespace, MS just mentions that it's part of the WinUI 3.0 preview...
https://docs.microsoft.com/en-us/windows/winui/api/microsoft.applicationmodel.resources?view=winui-3.0-preview

**Comment by GuildOfCalamity** [comment]
This is the error I see when loading the solution for the first time (from the designer, see screenshots below)

![XAML_issue1](https://user-images.githubusercontent.com/81915293/125128453-0ebab780-e0cc-11eb-915f-50ef698033d2.png)

![XAML_issue](https://user-images.githubusercontent.com/81915293/125127599-c0f17f80-e0ca-11eb-8861-ba27c0a8cd19.png)

**Comment by GuildOfCalamity** [comment]
Here is the only other clue I noticed (in App.xaml) ...

![XAML_error](https://user-images.githubusercontent.com/81915293/125176461-8870a500-e1a1-11eb-90f3-4a6c387eae36.png)

**Comment by GuildOfCalamity** [comment]
Also, when creating a blank project I do not see "**Blank App (WinUI in UWP)**" as noted in the book; maybe MS has changed the wording?

![WinUI New Project Options](https://user-images.githubusercontent.com/81915293/125176826-b6a3b400-e1a4-11eb-8201-830ad8c56d8f.png)

**Comment by alvinashcraft** [comment]
The UWP WinUI project types will not be available in the stable releases of Project Reunion (WindowsAppSDK). They will only remain in the preview releases for the foreseeable future. This was recently announced.
What you can probably do is create a WinUI Desktop project and import the existing code files (or follow the remaining steps in the book).

**Comment by GuildOfCalamity** [comment]
> 
> 
> The UWP WinUI project types will not be available in the stable releases of Project Reunion (WindowsAppSDK). They will only remain in the preview releases for the foreseeable future. This was recently announced.
> What you can probably do is create a WinUI Desktop project and import the existing code files (or follow the remaining steps in the book).

Would you recommend "Blank App" or just "App" from the menu? And did you find out any fix for the "Microsoft.ApplicationModel.Resources.winmd" reference issue?

**Comment by alvinashcraft** [comment]
The latest docs on WinUI recommend using the "Blank App" template. [Create new project - WinUI](https://docs.microsoft.com/en-us/windows/apps/winui/winui3/create-your-first-winui3-app)

I'm not seeing the issue with the winmd file that you have, but my VM still has old WinUI versions installed. I'm going to create a new Windows VM with the latest Visual Studio and WindowsAppSDK software so I can reproduce the issues that others will be encountering. I will probably create a new branch of this repo that will take the projects and target the latest WinUI bits. This will take a little time, but I'll keep you updated.

**Comment by GuildOfCalamity** [comment]
> 
> 
> The latest docs on WinUI recommend using the "Blank App" template. [Create new project - WinUI](https://docs.microsoft.com/en-us/windows/apps/winui/winui3/create-your-first-winui3-app)
> 
> I'm not seeing the issue with the winmd file that you have, but my VM still has old WinUI versions installed. I'm going to create a new Windows VM with the latest Visual Studio and WindowsAppSDK software so I can reproduce the issues that others will be encountering. I will probably create a new branch of this repo that will take the projects and target the latest WinUI bits. This will take a little time, but I'll keep you updated.

Thank you. I'll try starting with a blank app and importing the code. Is there an installer for the WindowsAppSDK?

**Comment by alvinashcraft** [comment]
The latest stable release of the WindowsAppSDK (formerly known as Project Reunion) is available [here](https://docs.microsoft.com/en-us/windows/apps/windows-app-sdk/stable-channel#version-08).

**Comment by GuildOfCalamity** [comment]
> 
> 
> The latest stable release of the WindowsAppSDK (formerly known as Project Reunion) is available [here](https://docs.microsoft.com/en-us/windows/apps/windows-app-sdk/stable-channel#version-08).

OK, I verified that this is what I installed before opening the projects.

**Comment by GuildOfCalamity** [comment]
I tried starting with the blank app, and the designer still won't render... strange.
![XAML_error](https://user-images.githubusercontent.com/81915293/125210128-c2a67900-e26b-11eb-86fd-369ec98b7ec7.png)

I uninstalled the WinUI3 (Project Reuion) and re-installed it into VS2019, but to no avail.
![InstalledComponents](https://user-images.githubusercontent.com/81915293/125210167-013c3380-e26c-11eb-8f91-427b9396102c.png)

Even the warnings seem bizare...
![WinUI3_error](https://user-images.githubusercontent.com/81915293/125210669-4c0b7a80-e26f-11eb-8151-76ebd175cc0d.png)

**Comment by GuildOfCalamity** [comment]
When fooling around with the NuGet package manager I've noticed that there is a **Microsoft.WinUI** and a **Microsoft.ProjectReunion.WinUI**, these two do not seem co-exist peacefully. Can you show me what your project references should look like?
![ProjectReunion](https://user-images.githubusercontent.com/81915293/125211753-97755700-e276-11eb-8369-f3d5f4e1a24f.png)

**Comment by jvsca** [comment]
I had the same issues. I typed the code, but at the end of Chapter 4 cannot add an item, **Windows.Current** always returned null.

I created a new project: **Blank App (Universal Windows)**, copied the code files and everything worked. Including the **WPF Editor**.

The NuGet packages are:

![image](https://user-images.githubusercontent.com/2821731/127399379-6d8cfc3a-389c-4c63-814e-757890bc098a.png)

The References are:

![image](https://user-images.githubusercontent.com/2821731/127399508-3af81447-f985-4c6e-9420-c9b158ad704a.png)

Also the book and the code for Chapter 4 are not the same, added to the constructor of **ItemDetailsViewModel**:

```
            PopulateLists();
            PopulateExistingItem(dataService);
            IsDirty = false;
```

as stated in the book (but missing in the code).

Hope this help.

FYI: Also working in Visual Studio 2022 Preview 2.1.
